source/lib/JMSLab/builders/build_latex.py

```diff
@@ -44,3 +46,72 @@ def add_call_args(self):
                                   os.path.normpath(self.log_file))
         self.call_args = args
         return None
+     
+    def add_bib_name(self, target):
+        if bool(target):
+            target_file = os.path.splitext(str(target[0]))[0]
+        else:
+            target_file = ''
+        self.bib_name = target_file
+        return None
+        
+    def check_bib(self, source):
+    
+        bibext = '.bib'
+        bib_file = ''
+        
+        if bool(source):
+            sources = misc.make_list_if_string(source)
+            for i in range(len(source)):
+                source_file = str(source[i])
+                if bibext in source_file:
```

Comment:
This matches `.bib` anywhere in `source_file`. It also might be prudent to do the check in lowercase, e.g. `source_file.lower().endswith(bibext)`